vortexasdk/endpoints/canal_transit.py

```diff
@@ -0,0 +1,212 @@
+
+            filter_direction: Direction that vessel is heading. Must be one of  [`'northbound'`, `'southbound'`],
+
+            filter_lock: Route taken through the canal. For the Panama canal, it must be one of [`'panamax'`, `'neopanamax'`],
+
+            filter_queue_arrival_time_min: The UTC date of when vessel arrived to the queue,
+
+            filter_queue_arrival_time_max: The UTC date of when vessel arrived to the queue,
+
+            filter_canal_entry_time_min: The UTC date of when vessel entered the canal,
+
+            filter_canal_entry_time_max: The UTC date of when vessel entered the canal,
+
+            filter_canal_exit_time_min: The UTC date of when vessel left the canal,
+
+            filter_canal_exit_time_max: The UTC date of when vessel left the canal,
+
+            filter_booked_time_min: The UTC date of planned time to enter the canal,
+
+            filter_booked_time_max: The UTC date of planned time to enter the canal,
+
+            filter_booked_status: Boolean to filter vessels that booked their slots. Can be `'true'` or `'false'`,
+
+            filter_voyage_status: A voyage status, or list of voyage statuses to filter on. Can be one of: `'ballast'`, `'laden'`,
+
+            updated_since: The UTC date of last record update time,
+
+            exclude_filter_vessels: A vessel ID, or list of vessel IDs to exclude,
+
+            exclude_filter_vessel_classes: A vessel class, or list of vessel classes to exclude,
+
+            exclude_filter_products: A product ID, or list of product IDs to exclude,
+
+            exclude_filter_origin: A geography ID, or list of geography IDs to exclude,
+
+            exclude_filter_destination: A geography ID, or list of geography IDs to exclude,
+
+            exclude_filter_charterer: A charterer ID, or list of charterer IDs to exclude,
+
+            exclude_filter_effective_controller: A effective controller ID, or list of IDs to exclude,
+
+        # Returns
+        `CanalTransitResults`, containing all the canal transit records matching the given search terms.
+
+        """
+
+        exclude_params: Dict[str, Any] = {
+            "filter_origin": convert_to_list(exclude_filter_origin),
+            "filter_destination": convert_to_list(exclude_filter_destination),
+            "filter_products": convert_to_list(exclude_filter_products),
+            "filter_vessels": convert_to_list(exclude_filter_vessels),
+            "filter_vessel_classes": convert_to_list(
+                exclude_filter_vessel_classes
+            ),
+            "filter_charterers": convert_to_list(exclude_filter_charterer),
+            "filter_effective_controllers": convert_to_list(
+                exclude_filter_effective_controller
+            ),
+        }
+
+        api_params: Dict[str, Any] = {
+            "filter_booked_time_max": to_ISODate(filter_booked_time_max)
+            if filter_booked_time_max
```

Comment:
This formatting is a bit weird, it took me a few secs looking at it to work out what was going on. Has this been passed through Black?

Reply:
yeah, I know, had the same feeling! Unfortunately that's how Black formatted it

Reply:
Roger that!